ha/core/event_manager/event_manager.py

```diff
@@ -57,10 +84,104 @@ def _validate_component(self, component: str) -> None:
         if component.lower() not in SUBSCRIPTION_LIST:
             raise InvalidComponent(f"Invalid component {component}, not part of subscription list.")
 
-    def _validate_events(self, events: list) -> None:
-        """
-        Raise error InvalidEvent if event is not valid.
+    def subscribe(self, component : str, events : list):
+        '''
+           Register all the events for the notification. It maintains
+           list of events registered by the components. It maps the event
+           name with component name and store it in the consul. \
+           Also, maps the component with event
 
+           Args:
+            component (str): Component Name
+            events (list): Events.
+        '''
+        Log.info(f'Received a subscribe request from {component} for event: {events}')
+        # self._validate_component(component)
+        self._validate_events(events)
+        if component: _create_message_type(component)
```

Comment:
```
        resource_type = events.resource_type
        states = events.states
```
all place this not much helpful
we can directly use events.resource_type events.states

Reply:
Done